File: dart/raindrops/lib/raindrops.dart
class Raindrops {
  static const MAP = {3: "Pling", 5: "Plang", 7: "Plong"};

  String convert(int i) => MAP.keys
      .where((e) => i % e == 0)
      .map((e) => MAP[e])
      .join("")
      .replaceAll(RegExp('^\$'), "$i");
}

File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

drops :: [(Int, String)]
drops = [(3, "Pling"), (5, "Plang"), (7, "Plong")]

convert :: Int -> String
convert n
  | null rain = show n
  | otherwise = rain
  where rain = concat [sound | (factor, sound) <- drops, n `rem` factor == 0]
